madmom/features/beats.py

```diff
@@ -554,6 +569,72 @@ def process(self, activations, **kwargs):
         # remove beats with negative times and return them
         return detections[np.searchsorted(detections, 0):]
 
+    def process_online(self, activations, reset=True, **kwargs):
+        """
+        Detect the beats in the given activation function for online mode.
+
+        Parameters
+        ----------
+        activations : numpy array
+            Beat activation function.
+        reset : bool, optional
+            Reset the BeatTrackingProcessor to its initial state before
+            processing.
+
+        Returns
+        -------
+        beats : numpy array
+            Detected beat positions [seconds].
+
+        """
+        # reset to initial state
+        if reset:
+            self.reset()
+        beats_ = []
+        for activation in activations:
+            # shift buffer and put new activation at end of buffer
+            buffer = self.buffer(activation)
+            # create an online interval histogram
+            histogram = self.tempo_estimator.online_interval_histogram(
+                activation, reset=False)
```

Comment:
Ah, I think this might have been a misunderstanding :)

For online mode, the `reset=reset` would be nicer for sure! What I was talking about was actually the single --online mode (sorry for not making this clear). This is where all the activations are passed at once to `process_online()`. And this is why there is a loop over all the activations inside this method. For online mode this loop is always called just once.

Another possibility would be, that for single --online `process_online()` gets called from the outside for each activation. Do you think that makes sense?